ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -139,7 +139,7 @@ data NodeArgs m peer blk = NodeArgs {
     , blockProduction     :: Maybe (BlockProduction m blk)
     , blockMatchesHeader  :: Header blk -> blk -> Bool
     , maxUnackTxs         :: Word16
-    , maxBlockBodySize    :: Word32
+    , maxBlockBodySize    :: Maybe Word32
```

Comment:
I think a separate type would be clearer, something like the following:
```
data MaxBlockBodySizeOverride
  = NoOverride
    -- ^ Use the maximum block size from the protocol parameters in the ledger.
  | MaxBlockBodySize Word32
    -- ^ Use the following maximum size for the block body
  | MaxBlockBody     Word32
    -- ^ Use the following maximum size for the whole block
```

Reply:
Yeah, I'd agree.